WalletWasabi.Fluent/ViewModels/CoinSelection/Core/TreeNode.cs

```diff
@@ -0,0 +1,25 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace WalletWasabi.Fluent.ViewModels.CoinSelection.Core;
+
+public class TreeNode
+{
+	public TreeNode(object value) : this(value, Enumerable.Empty<TreeNode>())
+	{
+	}
+
+	public TreeNode(object value, IEnumerable<TreeNode> children)
+	{
+		Value = value;
+		Children = children;
+	}
+
+	public object Value { get; }
+	public IEnumerable<TreeNode> Children { get; }
+
+	public override string? ToString()
+	{
+		return Value.ToString();
+	}
+}
```

Comment:
constructor allows null `Value`, but `ToString()` assumes not null. Might produce NRE

Add null check or make `ToString()` null-safe.